### PR TITLE
XS `Base64.decode` returns an `ArrayBuffer`, not `Uint8Array`

### PR DESCRIPTION
**Pull request by michaelfig**
This PR includes some changes I needed to do in order to debug trying to `ZipWriter.write(name, decodeBase64(str))`.

The solution is simple; just wrap the XS native decoder.

### CODE REVIEW

File: packages/base64/src/decode.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ import { monodu64, padding } from './common.js';
  * @param {string} string Base64-encoded string
  * @param {string} [name] The name of the string as it will appear in error
  * messages.
- * @returns {Uint8Array} decoded bytes
+ * @returns {Uint8Array} decoded bytes (array buffer on XS)
```
Comment (kriskowal) [inline packages/base64/src/decode.js:19]:
```suggestion
 * @returns {Uint8Array} decoded bytes
```

File: packages/base64/src/decode.js [hunk 2]
```diff
@@ -60,5 +60,18 @@ const jsDecodeBase64 = (string, name = '<unknown>') => {
   return data.subarray(0, j);
 };
 
+// The XS Base64.decode function is faster, but might return ArrayBuffer (not
+// Uint8Array).  Adapt it to our needs.
```
Comment (kriskowal) [inline packages/base64/src/decode.js:64]: Does XS return ArrayBuffer, Uint8Array, or either depending? I would find it weird if the signature varied. The conditional short might not be necessary.

**Review by kriskowal — APPROVED** [review]
How did this work before this change? Is there an XS bug in Base64.decode that makes the return type vary by size?

File: packages/base64/src/decode.js (hunk 2 above)
Comment (michaelfig) [reply packages/base64/src/decode.js:64]: It currently returns ArrayBuffer.  I don't want to be surprised if/when it used to return Uint8Array or will do so in the future.

**Pull request merged**